batchflow/models/tf/base.py

```diff
@@ -1648,74 +1648,100 @@ def output(self, inputs, predictions=None, ops=None, prefix=None, **kwargs):
         elif oper == 'proba':
-            self._add_output_proba(inputs, name, attr_prefix, device, **kwargs)
+            self._add_output_proba(inputs, name, attr_name, device, **kwargs)
         elif oper == 'labels':
-            self._add_output_labels(inputs, name, attr_prefix, device, **kwargs)
+            self._add_output_labels(inputs, name, attr_name, device, **kwargs)
         elif callable(oper):
-            self._add_output_callable(inputs, oper, None, attr_prefix, device, **kwargs)
+            self._add_output_callable(inputs, oper, None, attr_name, device, **kwargs)
 
-    def _add_output_identity(self, inputs, name, attr_prefix, device, **kwargs):
+    def _add_output_identity(self, inputs, name, attr_name, device, **kwargs):
         _ = kwargs
         x = tf.identity(inputs, name=name)
-        self.store_to_attr(attr_prefix + (name or ''), x, device)
+        self.store_to_attr(attr_name, x, device)
         return x
 
-    def _add_output_softplus(self, inputs, name, attr_prefix, device, **kwargs):
+    def _add_output_softplus(self, inputs, name, attr_name, device, **kwargs):
         _ = kwargs
         proba = tf.nn.softplus(inputs, name=name)
-        self.store_to_attr(attr_prefix + name, proba, device)
+        self.store_to_attr(attr_name, proba, device)
 
-    def _add_output_sigmoid(self, inputs, name, attr_prefix, device, **kwargs):
+    def _add_output_sigmoid(self, inputs, name, attr_name, device, **kwargs):
         _ = kwargs
         proba = tf.sigmoid(inputs, name=name)
-        self.store_to_attr(attr_prefix + name, proba, device)
+        self.store_to_attr(attr_name, proba, device)
 
-    def _add_output_proba(self, inputs, name, attr_prefix, device, **kwargs):
+    def _add_output_proba(self, inputs, name, attr_name, device, **kwargs):
         axis = self.channels_axis(kwargs['data_format'])
         proba = tf.nn.softmax(inputs, name=name, axis=axis)
-        self.store_to_attr(attr_prefix + name, proba, device)
+        self.store_to_attr(attr_name, proba, device)
 
-    def _add_output_labels(self, inputs, name, attr_prefix, device, **kwargs):
+    def _add_output_labels(self, inputs, name, attr_name, device, **kwargs):
         class_axis = self.channels_axis(kwargs.get('data_format'))
         predicted_classes = tf.argmax(inputs, axis=class_axis, name=name)
-        self.store_to_attr(attr_prefix + name, predicted_classes, device)
+        self.store_to_attr(attr_name, predicted_classes, device)
 
-    def _add_output_callable(self, inputs, oper, name, attr_prefix, device, **kwargs):
+    def _add_output_callable(self, inputs, oper, name, attr_name, device, **kwargs):
         _ = kwargs
         x = oper(inputs)
         name = name or oper.__name__
-        self.store_to_attr(attr_prefix + name, x, device)
+        self.store_to_attr(attr_name, x, device)
         return x
 
+    def _add_predictions(self, predictions, name, attr_name, **kwargs):
```

Comment:
It seems strange to always `stack` `head` outputs: what if you just wanted to have multiple outputs to process them individually in your `loss` function? That is, for example, the case for seismic noise attenuation models

Reply:
You can use outputs separately by names `predictions_0` `predictions_1` etc.